### PR TITLE
Use a random port for serving the GUI

### LINKED ISSUE #73: If GUI port is in use and not explicitly specified, pick a different port automatically

### PR DESCRIPTION
**Pull request by brianhelba**
Fixes #73.

### CODE REVIEW

**Comment by brianhelba** [comment]
@marySalvi PTAL.

File: imagedephi/utils/network.py [hunk 1]
```diff
@@ -12,3 +13,14 @@ async def wait_for_port(port: int, host: str = "127.0.0.1") -> None:
             writer.close()
             await writer.wait_closed()
             return
+
+
+def unused_tcp_port() -> int:
+    with socket.socket() as sock:
+        sock = socket.socket()
```
Comment (marySalvi) [inline imagedephi/utils/network.py:20]: Just for my own understanding, wouldn't sock already be `socket.socket()` from line 19? I am unsure why line 20 is necessary.

**Review by marySalvi — APPROVED** [review]
LGTM Tested on windows and ubuntu

File: imagedephi/utils/network.py (hunk 1 above)
Comment (brianhelba) [reply imagedephi/utils/network.py:20]: You're right. I think I was testing something and accidently committed this line. I'll fix it.

**Review by marySalvi — APPROVED** [review]

**Pull request merged**